verifiers/internal/gha/verifier.go

```diff
@@ -71,15 +71,8 @@ func verifyEnvAndCert(env *dsse.Envelope,
 	// There is a corner-case to handle: if the verified builder ID from the cert
 	// is a delegator builder, the user MUST provide an expected builder ID
 	// and we MUST match it against the content of the provenance.
-	if byob {
-		if builderOpts.ExpectedID == nil || *builderOpts.ExpectedID == "" {
-			// NOTE: we will need to update the logic here once our default trusted builders
```

Comment:
why not keep this code and set ExpectedBuilderID if builderOpts.ExpectedID is empty?
```
provenanceOpts.ExpectedBuilderIDPath = "https://github.com/slsa-framework/slsa-github-generator/.github/workflows/"
```

Reply:
The way the structure of code is, we need provenance to check expected path and verify against `BuilderID` such that we dont get the empty builder error. That is why it got moved into `verifyProvenance`!